### PR TITLE
Add custom exception

### PR DESCRIPTION
**Pull request by Chemaclass**
A little refactoring adding a custom exception `MissingApiCredentialsException `

### CODE REVIEW

**Review by pscheit — COMMENTED** [review]
Hey!

thanks for the PR! I was wondering two things

1. were the unrelated code style changes introduced cause you ran the php cs fixer? Our codestyle imho doesnt require to break long lines.
2. Why do we need a Custom Exception in the console command?

**Comment by Chemaclass** [comment]
Hi, @pscheit ! thanks for the feedback. Regarding those questions:
1.  Well, that's kind of a mix. But mostly, I thought it would be way easier to read the line if we break the long lines into multiline; at least for readability purposes.
2. The custom exception is to simplify the readability by using a named constructor for it, so the message of the exception is a hidden in the detail. It might not be required or adding huge value/impact, but I thought it would be worth showing it as an example to see your thoughts too :)

**Comment by Chemaclass** [comment]
I was thinking, as next step, I would like to try to refactor it to get rid of the testing public method
```php
    /** @test */
    public function setPackagistApiClient(PackagistApiClient $client): void
    {
        $this->packagistApiClient = $client;
    }
```
and, for example, consider injecting this via the constructor, instead of using a public setter specificaly for the test. This way, we can inject and mock the interface in the test, while using the real instanciation on the prod code - moving the logic from `getPackagistApiClient()` up to `bin/console`, for example.

File: src/Command/ListCommand.php
Comment (pscheit) [inline src/Command/ListCommand.php:1]: Can you revert those multi lines to be just single line as before? Thanks

**Review by pscheit — CHANGES_REQUESTED** [review]
I still dont see much value in adding those exception, but they don't hurt either :)

So happy to merge if the you revert the unrelated changes

**Comment by pscheit** [comment]
I mean sure, if you want to, go for it.

**Comment by Chemaclass** [comment]
Good for me. Do you mean the option and/or the array_map usage? I would personally prefer to keep the array_map calls in multiline (as they read it simpler). Would that be ok, or would you prefer to rollback both of them and move this to a different PR to unblock the other changes?

**Comment by pscheit** [comment]
Please, just provide the changes with the custom exception and the phpunit changes.
Remove all other style changes.

I wouldn't suggest working on this repo further and sending more PRs. This is really just a small demo which might not be the nicest code but really is enough to demonstrate.

Thanks for your efforts!

**Comment by Chemaclass** [comment]
@pscheit can you check it again? thanks!

**Review by pscheit — APPROVED** [review]
thanks

**Pull request merged**